File: src/main/kotlin/Q_AnonymousInnerClass.java
/**
 * Anonymous Class
 *
 * Interface obj = new Interface(){//code}
 */

interface Playable {
    void play();
}

class Q_AnonymousInnerClass {

    public static void main(String[] args) {
        //without lambadas
        Playable obj1 = new Playable() {
            @Override
            public void play() {
                System.out.println("Lets play Cricket");
            }
        };

        //Java 8
        Playable obj2 = () -> System.out.println("Lets play Football");

        //Calling method
        obj1.play();
        obj2.play();
    }
}

File: src/main/kotlin/Q_AnonymousInnerClass.kt
/**
 *  Anonymous Class
 *
 *
 *  obj:Interface = object:Interface{//code}
 *
 */

interface Playable2 {
    fun play()
}

fun main() {
    println("\n=========== ProdBug ==========")

    var obj: Playable2 = object : Playable2 {
        override fun play() {
            println("Lets play Basketball")
        }
    }

    println(obj.play())

}

